python/ray/autoscaler/_private/autoscaler.py

```diff
@@ -388,6 +399,43 @@ def terminate_scheduled_nodes(self):
         self.nodes_to_terminate = []
         self.update_worker_list()
 
+    def drain_nodes_via_gcs(self):
+        node_ips = [
+            self.provider.internal_ip(node_id)
+            for node_id in self.nodes_to_terminate
+        ]
+        raylet_ids_to_drain = {
+            self.load_metrics.raylet_id_by_ip[ip]
+            for ip in node_ips
+        }
+
+        logger.info(f"Draining raylets with ids {raylet_ids_to_drain}.")
+        request = gcs_service_pb2.DrainNodeRequest(drain_node_data=[
```

Comment:
Maybe we should add try catch here? (The rpc doesn’t exist in older versions)

Reply:
I think this request class exists on the client (autoscaler) side for Ray versions since your commit. So we should be fine.
The thing to guard against is the server not knowing what to do with the request (grpc NotImplemented error).